qemu/target-i386/cpu.h

```diff
@@ -699,7 +699,7 @@ typedef enum {
 
 typedef struct SegmentCache {
     uint32_t selector;
-    target_ulong base;
+    uint64_t base;   /* handle 32 or 64 bit CPUs */
```

Comment:
That will work if we copy the definition of x86_mmr from x86.h into cpu.h or unicorn.c (your choice) so that we can cast appropriately when we receive the void pointer in uc_reg_read/write

I am fine with that if that is what you want to do.